modules/setting/setting.go

```diff
@@ -613,7 +621,7 @@ func NewContext() {
 	OfflineMode = sec.Key("OFFLINE_MODE").MustBool()
 	DisableRouterLog = sec.Key("DISABLE_ROUTER_LOG").MustBool()
 	StaticRootPath = sec.Key("STATIC_ROOT_PATH").MustString(AppWorkPath)
-	AppDataPath = sec.Key("APP_DATA_PATH").MustString(path.Join(AppWorkPath, "data"))
+	AppDataPath = sec.Key("APP_DATA_PATH").MustString("data")
```

Comment:
This should not be changed

Reply:
I need the followings:
```go
AvatarUploadPath = "data/avatars/"
LFS.ContentPath	 = "data/lfs/"
AttachmentPath	 = "data/attachments/"
RepositoryAvatarUploadPath = "data/repo-avatars/"
```

Should I use:
`LFS.ContentPath = sec.Key("LFS_CONTENT_PATH").MustString(filepath.Join("data", "lfs/"))`
instead of:
`LFS.ContentPath = sec.Key("LFS_CONTENT_PATH").MustString(filepath.Join(AppDataPath, "lfs"))`

Reply:
Problem is that this is just defaults. User could have specified completely different paths in their app.ini for ex:
```ini
AvatarUploadPath = "/srv/avatars/upload/"
LFS.ContentPath	 = "/srv/lfs/"
AttachmentPath	 = "/srv/attachments/"
RepositoryAvatarUploadPath = "/srv/avatars/repo-upload/"
```

And in such configuration your code will not work for local storage anymore

Reply:
It will work for either local or any other cloud storage.
By default I set it to local storage. https://github.com/go-gitea/gitea/blob/16ac0bb98f889f15628e1779889fd0b689927f2a/modules/setting/file_storage.go#L10-L23
If user wants to use any other cloud storage or other local storage, they just have to specify it in their `app.ini`

Users are free to provide their own upload path like you suggested and it will work fine.

Reply:
I meant it should work for local storage same as it was before without changing app.ini

Reply:
Yes, it will work as it was before. Just the files will be stored in the directory specified in `FileStorage.BucketURL`.

So, what do you suggest about https://github.com/go-gitea/gitea/pull/7795#discussion_r311939134 ?

Reply:
So I've checked out this PR and ran it with my usual test settings:

```
[server]
SSH_DOMAIN       = ....
DOMAIN           =  ....
HTTP_PORT        =  ....
PROTOCOL         = http
ROOT_URL         =  ....
OFFLINE_MODE     = false
APP_DATA_PATH    = /mnt/external/my_test_data
```

Then I've changed my avatar. I _expected_ the avatar to be created at:
```
/mnt/external/my_test_data/avatars/a9050bf53d1005dfe5398ce9fa5b07aa
```

Instead, with this PR, the avatar was **incorrectly** created at:
```
/home/giteatest/avatars/a9050bf53d1005dfe5398ce9fa5b07aa
```

As you can see, with this PR Gitea completely ignored the value of my `APP_DATA_PATH`. This is the breaking change we've been talking about. The code must behave **like it did before this PR** if the user doesn't change their `app.ini`. Users that upgrade to this must not notice any difference, unless they actively decide to use the new bucket storage capability.

I hope that this example helps.

Reply:
Hey @guillep2k, thanks for taking the time for testing.

If `AVATAR_UPLOAD_PATH` is set either relative or absolute, it ignores `APP_DATA_PATH` as it was ignored previously. https://github.com/go-gitea/gitea/blob/master/modules/setting/setting.go#L850-L854

But if `AVATAR_UPLOAD_PATH` is not specified, then it uses `APP_DATA_PATH`+"avatars" to read or store the avatars.

Either way if the `AvatarUploadPath` is relative then the whole path to store the avatars is `AppWorkPath`+`AvatarUploadPath`.
If the path is absolute the avatars is stored to `AvatarUploadPath`.
It works like it did previously.

I've attached the scenarios here : https://github.com/go-gitea/gitea/blob/e29ba594b0/modules/storage/storage.go#L124-L154

May be there was some errors in my previous commit but I refactored the code and I hope it works like it did before this pr.

**Please, pull again and check if it's now okay. Thanks again.**